devito/arch/compiler.py

```diff
@@ -761,7 +763,9 @@ def __init__(self, *args, **kwargs):
         platform = kwargs.pop('platform', configuration['platform'])
         language = kwargs.pop('language', configuration['language'])
 
-        if language == 'openmp':
+        # Since OneAPI 2023.2.0 (clang17 underneath), we can use '-qopenmp'
+        # for all our tests
+        if self.version >= Version('17.0.0') and language == 'openmp':
```

Comment:
isn't this doing the opposite of what the comment says?

also, if you are to change this (maybe not, maybe I misunderstood...), could you drop "for all our tests" which doesn't mean much? what you should have said is that earlier versions have an OpenMP bug. The bug is in their compilers, it doesn't have anything to do with our tests (I mean, not directly)

Reply:
damn sorry, you are right